### PR TITLE
Added http endpoint for metadata

### PR DESCRIPTION
**Pull request by parauliya**

### CODE REVIEW

File: grpc_server.go [hunk 1]
```diff
@@ -31,16 +28,6 @@ type exportedHardware struct {
...
 }
 
-var active_subscription = promauto.NewGauge(prometheus.GaugeOpts{
-	Name: "hegel_active_subscriptions",
-	Help: "Number of active hegel subscribers",
-})
-
-var active_cacher_subscription = promauto.NewGauge(prometheus.GaugeOpts{
-	Name: "hegel_active_cacher_subscriptions",
-	Help: "Number of active hegel subscriptions to cacher",
```
[the remark this answers is not in the document]
Comment (parauliya) [reply grpc_server.go]: Done

File: grpc_server.go [hunk 2]
```diff
@@ -12,9 +12,6 @@ import (
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/peer"
 	"google.golang.org/grpc/status"
-
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promauto"
```
[the remark this answers is not in the document]
Comment (parauliya) [reply grpc_server.go]: Done

File: go.mod [hunk 3]
```diff
@@ -1,10 +1,10 @@
 module github.com/packethost/hegel
 
-go 1.13
+go 1.12
```
[the remark this answers is not in the document]
Comment (parauliya) [reply go.mod]: Done

File: grpc_server.go [hunk 4]
```diff
@@ -31,14 +31,14 @@ type exportedHardware struct {
 	BondingMode                        int                      `json:"bonding_mode"`
 }
 
-var active_subscription = promauto.NewGauge(prometheus.GaugeOpts{
-	Name: "hegel_active_subscriptions",
+var activeSubscription = promauto.NewGauge(prometheus.GaugeOpts{
+	Name: "hegel_activeSubscriptions",
```
Comment (nathangoulding) [inline grpc_server.go]: This doesn't seem like a necessary change

File: grpc_server.go [hunk 5]
```diff
@@ -31,14 +31,14 @@ type exportedHardware struct {
...
 	Help: "Number of active hegel subscribers",
 })
 
-var active_cacher_subscription = promauto.NewGauge(prometheus.GaugeOpts{
-	Name: "hegel_active_cacher_subscriptions",
-	Help: "Number of active hegel subscriptions to cacher",
+var activeTinkSubscription = promauto.NewGauge(prometheus.GaugeOpts{
+	Name: "hegel_activeTinkSubscriptions",
```
Comment (nathangoulding) [inline grpc_server.go]: I would keep snake case here

File: main.go [hunk 6]
```diff
@@ -130,93 +121,36 @@ func main() {
...
-			cancel()
-
-			isCacherAvailableMu.Lock()
-			isCacherAvailable = isCacherAvailableTemp
-			isCacherAvailableMu.Unlock()
-
-			logger.With("status", isCacherAvailableTemp).Debug("tick")
-		}
-
-	}()
```
Comment (nathangoulding) [inline main.go]: Why was this removed?

File: main.go [hunk 7]
```diff
@@ -130,93 +121,36 @@ func main() {
...
-	b, err := json.Marshal(&res)
+	err = grpcServer.Serve(lis)
 	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
+		logger.Fatal(err, "Failed to serve  grpc")
 	}
 
-	if !isCacherAvailableTemp {
-		w.WriteHeader(http.StatusInternalServerError)
-	}
```
Comment (nathangoulding) [inline main.go]: If hegel can't connect to the underlying data store (either cacher or tink), this seems like something we would still want to handle.

File: main.go [hunk 8]
```diff
@@ -54,53 +53,45 @@ func envString(key string, def string) (val string) {
...
-
-	grpcConfig, err := packetgrpc.ConfigFromEnv()
-	if err != nil {
-		logger.Error(err, "Failed to get config")
-		panic(err)
-	}
+	defer l.Close()
 
-	grpcAddr, err := grpcConfig.BindAddress()
+	port, err := strconv.Atoi(env.Get("GRPC_PORT", "42115"))
```
Comment (nathangoulding) [inline main.go]: use `net.parsePort`

File: main.go [hunk 9]
```diff
@@ -130,93 +121,36 @@ func main() {
...
-	http.HandleFunc("/_packet/healthcheck", healthCheckHandler)
-	http.HandleFunc("/_packet/version", versionHandler)
-	logger.With("port", *metricsPort).Info("Starting http server")
-	go func() {
-		err := http.ListenAndServe(fmt.Sprintf(":%d", *metricsPort), nil)
-		if err != nil {
-			logger.Error(err, "failed to serve http")
-			panic(err)
-		}
-	}()
```
Comment (nathangoulding) [inline main.go]: Why was this changed around?

File: grpc_server.go (hunk 4 above)
Comment (parauliya) [reply grpc_server.go]: Yes, but I did this because of naming convention in golang.

File: main.go (hunk 8 above)
Comment (parauliya) [reply main.go]: ok

File: main.go (hunk 6 above)
Comment (parauliya) [reply main.go]: I didn't see any use case for it so I removed to keep the code simple.
could you please brief me about the use case for it?

File: main.go (hunk 9 above)
Comment (parauliya) [reply main.go]: Because I moved all the http calls in a new file called http_handler.go

File: main.go (hunk 7 above)
Comment (parauliya) [reply main.go]: All of these http calls calls and function are there as it was before. They are just moved from main.go to http_handle.go

File: grpc_server.go (hunk 5 above)
Comment (parauliya) [reply grpc_server.go]: Changed to snake cases here.

File: grpc_server.go (hunk 4 above)
Comment (parauliya) [reply grpc_server.go]: Reverted it.

File: main.go (hunk 8 above)
Comment (parauliya) [reply main.go]: I am not able to use this function since this is not exported.

File: main.go (hunk 6 above)
Comment (parauliya) [reply main.go]: Revert back.
